### PR TITLE
feat: Add resizeable property to text and richText fields, and apply to image altText

### PR DESCRIPTION
**Pull request by jonathonherbert**
## What does this change?

Adds an `isResizeable` property to text and richtext fields, to allow both sorts to be resizeable by users in the usual way. This PR also applies this to our image alt text, to reflect the current behaviour of the old image element:

resize

There's one decision I've made here that's worth mentioning – when an input is resizeable, we default its initial size to the number of rows specified in the `rows` property. I thought this made a sort of heuristic sense, as whenever we make something resizeable, we're likely expecting lots of content in a limited space – but others may disagree! We can use `maxRows` to ensure that users can't expand inputs indefinitely if required.

Incidentally, I was dusting off the HTML5 drag and drop API docs before I found out this was possible with CSS – phew! 😅 

## How to test

- Drag is difficult to test in integration directly, so I've added integration tests to check the appropriate css property is set, and to check that the height behaviour above is working correctly.
- Try the image element shown in the GIF. Does it work as expected?

### CODE REVIEW

File: src/plugin/fieldViews/TextFieldView.ts [hunk 1]
```diff
@@ -137,18 +149,26 @@ export class TextFieldView extends ProseMirrorFieldView {
...
 
         if (maxRows) {
-          (this.innerEditorView.dom as HTMLDivElement).style.maxHeight = `${
+          const maxHeightPx = `${
             parseInt(lineHeight, 10) * maxRows + parseInt(paddingTop)
           }px`;
-          (this.innerEditorView.dom as HTMLDivElement).style.overflowY =
-            "scroll";
+          domElement.style.maxHeight = maxHeightPx;
+          domElement.style.overflowY = "scroll";
```
Comment (dskamiotis) [inline src/plugin/fieldViews/TextFieldView.ts:171]: Great simple implementation with HTML 5, as not aware of this and expected some fancy js!

File: src/elements/image/ImageElement.tsx [hunk 2]
```diff
@@ -61,9 +61,10 @@ export const createImageFields = ({
 }: MainImageProps) => {
   return {
     alt: createTextField({
-      rows: 2,
+      rows: 4,
```
Comment (dskamiotis) [inline src/elements/image/ImageElement.tsx:64]: Increasing the default a result of 
`something resizeable, we're likely expecting lots of content in a limited space`?

**Review by dskamiotis — APPROVED** [review]
Really useful addition @jonathonherbert, and elegant simple solution with the HTML 5.

Tested successfully and working as expected locally in Composer, linking your branch via yalc.
One possible side effect, it also makes the width expand the bounds of the element body area.

**Comment by rhystmills** [comment]
Amazing, I never knew that property existed - this will save me a lot of pain in the future, when I inevitably would have sunk many hours trying to build a custom React component.

I suspect a few of our other elements would benefit from something like this - off the top of my head, the code field in the code element and the preview in the embed element,.

**Comment by dskamiotis** [comment]
@rhystmills good thinking, definitely will be applicable and easily implemented in some current/future elements 👍

File: src/elements/image/ImageElement.tsx (hunk 2 above)
Comment (jonathonherbert) [reply src/elements/image/ImageElement.tsx:64]: Yeah, precisely – but maybe it's now a little big?

Comment (dskamiotis) [reply src/elements/image/ImageElement.tsx:64]: I think we could go for 3 as default middle ground :)
Alternatively I don't think it makes a huge difference IMO

Comment (jonathonherbert) [reply src/elements/image/ImageElement.tsx]: Tested inn Composer, 2 is better – thanks for mentioning 👍

**Comment by jonathonherbert** [comment]
@rhystmills good shout, I've added that and tested in Composer 👍

**Pull request merged**